File: Data Structures & Algorithms/jump-game/submission-0.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        """
            lets do brute force first
            think through the solution
        """

        def helper(idx,nums: List[int]) -> bool:
            if idx >= len(nums)-1:
                return True
            
            ans = False
            for i in range(1, nums[idx]+1):
                ans |= helper(idx + i, nums)

            return ans
        

        return helper(0,nums)

File: Data Structures & Algorithms/count-subsequences/submission-0.cpp
class Solution {
public:
    int recurse(string s,string t,int i,int j){
        if(j == t.size()){  
            return 1;
        }

        if(i == s.size()){  
            return 0;
        }
        
        int ans = 0;
        if(s[i] == t[j]){
            ans += recurse(s,t,i+1,j+1);
        }

        ans += recurse(s,t,i+1,j);

        return ans;

    }
    int numDistinct(string s, string t) {
        /*
            c   a   a   a   t
                        |
            c   a   t
                    |
        */
        return recurse(s,t,0,0);
    }
};
